include/cadmium/modeling/dynamic_models_helpers.hpp

```diff
@@ -79,6 +103,37 @@ namespace cadmium {
             };
             for_each<BST>(bs, add_messages_to_map);
         }
+
+        bool is_in(const std::type_index& port, const ports_vector& ports) {
+            return std::find(ports.cbegin(), ports.cend(), port) != ports.cend();
+        }
+
+        bool valid_ic_links(const models_map& models, const links_vector& ic) {
+            for (auto & link : ic) {
+                if (link.size() != 4 || models.find(link[0]) == models.cend() || models.find(link[3]) == models.cend()) {
+                    return false;
+                }
+            }
+            return true;
+        }
+
+        bool valid_eic_links(const models_map& models, const ports_vector& input_ports, const links_vector& eic) {
+            for (auto & link : eic) {
```

Comment:
Maybe some of these validation rules are simpler to write using all_of/none_of/any_of

Reply:
Ok, I'll check if that's the case

Reply:
Done.